adapter_pipelines/submit.wdl

```diff
@@ -142,10 +142,19 @@ task stage_files {
 
     # Stage the files
     files=( ${sep=' ' files} )
+
+    # Check if any files were already uploaded to the staging area
+    uploaded=($(hca upload list))
+
     for f in "$${lb}files[@]${rb}"
     do
-      echo "hca upload file $f"
-      hca upload file $f
+      file_path=($(echo "$f" | tr '/' '\n'))
+      file_name="$${lb}file_path[$${lb}#file_path[@]${rb} -1]${rb}"
+      is_uploaded=$(echo "$${lb}uploaded[@]${rb}" | grep -o $file_name | wc -w)
```

Comment:
This is clever! But I think it would be clearer to write a bit of Python code that could filter the list down to just files that haven't been uploaded already, and then pass the filtered list to this loop instead. There would be an added advantage that you could write unit tests for that Python code, whereas the bash can't really be tested without running the WDL.